### PR TITLE
DOC: added examples to docstring of dirichlet class

### PR DESCRIPTION
**Pull request by jeannefukumaru**

### CODE REVIEW

**Comment by jeannefukumaru** [comment]
This is my first pull request to an open-source project. Any pointers on how to improve are much appreciated!

**Comment by ilayn** [comment]
Hi @jeannefukumaru Thanks a lot for taking the time to prepare this. You are doing quite good so far. A few details to iron out: 

 - In the docs, numpy is implicit to all examples, you can skip `import numpy as np` line
 - You might want to regroup similar items and decide on whether the explanations should be text or inline comment. Take [ndimage.convolve](http://scipy.github.io/devdocs/generated/scipy.ndimage.convolve.html) as a random example. There are parts as explanations and parts as code (you can look at its source code how to do that). I think we can be a bit more economical here.  
  - you can skip the assignments to `d` just to print. Hence instead of 
    ```python
    >>> d = dirichlet.pdf(quantiles, alpha)
    >>> d  
    0.2843831684937255
    ```
    you can directly type
    ```python
    >>> dirichlet.pdf(quantiles, alpha)
    0.2843831684937255
    ```
 - I'll leave it up to your decision but either make it text or inline code comment for the initial 3 lines. Note that you can also use inline comments if it is short enough
    ```python
    >>> dirichlet.rvs(alpha, size=2)  # get random samples from dirichlet
    array([[3.13387474e-02, 1.07631504e-01, 8.61029749e-01],
       [5.28822173e-05, 3.21909147e-01, 6.78037971e-01]])
    ```

**Comment by ilayn** [comment]
You can also track what went wrong in the CI suite, for this case these are the documentation errors detected 

https://travis-ci.org/scipy/scipy/jobs/436477657#L1369

**Comment by ilayn** [comment]
Thanks @jeannefukumaru, almost there, two minor items remaining :) 

Currently it renders like this https://9221-1460385-gh.circle-artifacts.com/0/html-scipyorg/generated/scipy.stats.dirichlet.html#scipy.stats.dirichlet 

that is because it needs empty lines also *after* the text blocks in the examples.  And also this line

https://github.com/scipy/scipy/blob/e7807260cf28d53b101c5a021256bdb6cdc88cba/scipy/stats/_multivariate.py#L1385

is too long for 79 character limit. Otherwise it is good to go.

**Comment by jeannefukumaru** [comment]
Thanks for the detailed feedback! It was very helpful. I've committed the edits you suggested

File: scipy/stats/_multivariate.py [hunk 1]
```diff
@@ -1368,6 +1368,40 @@ class dirichlet_gen(multi_rv_generic):
...
+    >>> dirichlet.mean(alpha)  # get the mean of the distribution
+    array([0.01960784, 0.24509804, 0.73529412])
+    >>> dirichlet.var(alpha) # get variance
+    array([0.00089829, 0.00864603, 0.00909517])
+    >>> dirichlet.entropy(alpha)  # calculate the differential entropy
+    -4.3280162474082715
+
+    We can also return random samples from the distribution
+
+    >>> dirichlet.rvs(alpha, size=1)  # may vary
```
[the remark this answers is not in the document]
Comment (ilayn) [reply scipy/stats/_multivariate.py]: I agree that is much better. 

@jeannefukumaru Would you like to add those `random_state` keywords from [here](http://scipy.github.io/devdocs/generated/scipy.stats.rv_continuous.rvs.html). I don't want to spoil the fun by jumping the gun.

**Comment by jeannefukumaru** [comment]
Apologies for the delay and thanks for your patience  I've added the `random_state` keyword to `dirichlet.rvs`. 

After looking at the details for why the CI builds are failing, it seems like python 2.7 and python 3.6 are two environments that are throwing the errors. Looks like the `git checkout -qf FETCH_HEAD` is giving a service error
screen shot 2018-10-24 at 3 48 28 pm

**Comment by ilayn** [comment]
Travis Python 3.6 failure seems like a real one. The expected value for size = 2 is wrong. I get the same value as Travis CI reports

```python
>>> dirichlet.rvs(alpha, size=1, random_state=1)
array([[0.00766178, 0.24670518, 0.74563305]])

>>> dirichlet.rvs(alpha, size=2, random_state=2)
array([[0.01639427, 0.1292273 , 0.85437844],
       [0.00156917, 0.19033695, 0.80809388]])
```

**Comment by ilayn** [comment]
Since this is almost done, I've rebased it. If CI is happy it can go in.

**Comment by ilayn** [comment]
Barely made it. Thanks @jeannefukumaru

**Pull request merged**